### ISSUE TITLE
learn-tape given too much time - add fizz buzz workshop where students write their own tests

### ISSUE THREAD

**Issue by finnhodgkin** [foundersandcoders/master-reference#434]
We allocated two hours for the dwyl [learn-tape](https://github.com/dwyl/learn-tape) workshop but everyone finished it in an hour and the stretch goals in another half hour. @oliverjam and I replaced the final 30 mins with an impromptu half hour workshop on git amend & reset but it definitely wasn't ideal. Something test-related would be way more appropriate.

On the other hand the Roman Numerals code along ran over by a quarter of hour and still wasn't finished (css wasn't covered).

**Comment by finnhodgkin** [comment]
Some solutions:
+ Add more material to Learn-Tape
+ Allocate more time to Roman Numerals and less to Learn-Tape
+ Add a short challenge / workshop with the format of 'here is challenge (something like a kata?), complete using TDD'

**Comment by eliascodes** [comment]
@finnhodgkin option 2 is zero extra work. Would be ideal if you feel like the Roman numerals WS can be split over lunch.

**Comment by oliverjam** [comment]
@eliascodes We took the full two hours and even overran a tiny bit for the Roman Numerals codealong, and didn't even get to write any CSS. It might be better to extend this and almost make it a mini-project

**Comment by eliascodes** [comment]
@oliverjam I was thinking 
11-12: learn-tape
12-13: Roman pt1
14-16: Roman pt2

**Comment by jsms90** [comment]
It sounded to me like the issues with day 1 are about more than just timing?

Given the design of the workshop (the code for the tests is provided at each step), it's entirely likely that students will copy-paste everything in `learn-tape` & therefore not write a single test themselves before their projects.

This may or may not be why the workshop took so little time to complete. Students are not required to think for themselves about what a good test would be.

@oliverjam @finnhodgkin What are the "learning outcomes" of this portion of the day?
+ Students learn how to use npm & install tape via the 1st 30 mins of the day.
+ Students learn about TDD and learn the tape syntax during a very comprehensive code-along
+ What is `learn-tape` adding?

Given that we seem to have time in this week, where we could add something that would get students writing their own tests, I would go with @finnhodgkin's 3rd suggestion
> Add a short challenge / workshop with the format of 'here is challenge (something like a kata?), complete using TDD'

Less effort:
+ the fizzbuzz workshop that already exists (substitute in tape for QUnit)

More effort:
+ pick a (thoroughly vetted) codewars kata
+ write a new workshop e.g. the tape version of `dwyl/learn-tdd` https://github.com/dwyl/learn-tdd/issues/90

@BartBucknill @morkeltry I would be interested to hear your thoughts

**Comment by BartBucknill** [comment]
As Jen wrote, really just had us copying and pasting, or at best copying, code. 
The code along was great, and after that most people would probably be able to have a go at writing their own tests for a TDD approach to a coding challenge.
I like the sound of amending the existing fizzbuzz workshop for use with tape.

**Comment by dangerdak** [comment]
I also like the workshop idea, as the learn-tape workshop and code-along were great but some more involved learning would really help solidify the stuff we went over, and prepare us for a test-driven project. A short challenge/workshop where the tests aren't given to you sounds perfect. Learning by doing :+1:

**Comment by jsms90** [comment]
@RQ16 @m4v15 I just want you guys to be aware of this

**Comment by oliverjam** [comment]
Unless there are any objections I'll convert the [FizzBuzz workshop](https://github.com/foundersandcoders/fizzbuzz) to Tape in the morning

**Comment by jsms90** [comment]
@oliverjam @m4v15 @RQ16 Looks like the fizz buzz workshop has been converted to use tape https://github.com/foundersandcoders/fizzbuzz/issues/5

Who is updating the week schedule for tomorrow in Nazareth?

**Comment by m4v15** [comment]
Is it not worth us just seeing how it goes in Nazareth tomorrow? 

So stick with the schedule as it is for now, but if we find we have a lot of time after the learn tape, then we'll either do fizz buzz or start the roman numerals. We don't need to commit to officially changing the schedule or anything right now.

 Also like @eliascodes points out, do we think splitting the code along over lunch is fine? I can see it maybe being a bit disruptive, but I guess we won't know until we try it, it'll probably be alright.

**Comment by oliverjam** [comment]
@m4v15 How did the first day go? What schedule did you end up going with?

**Comment by RQ16** [comment]
@oliverjam, the first day went well. We dedicated 2 hours for the tape workshop. We did the first half an hour of the Roman Numeral challenge as a code-along and then let the students pair on it for 40 minutes. In the last 20 minutes we had someone cast the screen and explain what they did.

**Comment by oliverjam** [comment]
@RQ16 Awesome, that sounds good. Glad it's going well!

**Comment by m4v15** [comment]
Yeah so we found that we were going through the code along pretty slowly and were aware that students weren't writing their own code - I would ask them to navigate me, but it would only be 1 or 2 of the same students every time. So to counter this we let them pair off and do Roman Numerals themselves - this went really well actually and gave them good experience of TDD. I made it clear that the important part was writing tests. At the end I asked if they wanted to continue themselves or go through the solution together, they all wanted a bit more time, so then in the last 5-10 minutes I showed *a* solution and directed them towards Ollie's code along for a more thorough explanation.
